### PR TITLE
Fixes floordiv dunder registrations

### PR DESCRIPTION
**Pull request by mruberry**
floordiv was missing a couple dunder registrations, which was causing __ifloordiv__ to not be called when it should. This adds the appropriate registrations and adds a test verifying that the inplace dunders are actually occuring inplace.

### CODE REVIEW

File: tools/autograd/templates/python_variable_methods.cpp [hunk 1]
```diff
@@ -854,7 +854,9 @@ PyMethodDef variable_methods[] = {
   {"__isub__", (PyCFunction)(void(*)(void))TypeError_to_NotImplemented_<THPVariable_sub_>, METH_VARARGS | METH_KEYWORDS, NULL},
   {"__div__", (PyCFunction)(void(*)(void))TypeError_to_NotImplemented_<THPVariable_div>, METH_VARARGS | METH_KEYWORDS, NULL},
   {"__truediv__", (PyCFunction)(void(*)(void))TypeError_to_NotImplemented_<THPVariable_div>, METH_VARARGS | METH_KEYWORDS, NULL},
+  {"__floordiv__", (PyCFunction)(void(*)(void))TypeError_to_NotImplemented_<THPVariable_floor_divide>, METH_VARARGS | METH_KEYWORDS, NULL},
```
Comment (gchanan) [inline tools/autograd/templates/python_variable_methods.cpp:857]: do you know why we manually bind these but some things in native_functions are registered there?  Like __and__ for example.  Maybe it has to do with TypeError_to_NotImplemented_ but I don't really see why those wouldn't apply to native_functions.

File: test/test_torch.py [hunk 2]
```diff
@@ -5322,6 +5322,18 @@ def add_neg_dim_tests():
 class TestTorchDeviceType(TestCase):
     exact_dtype = True
 
+    # Verifies that the inplace dunders (like idiv) actually are in place
+    @onlyOnCPUAndCUDA
+    def test_inplace_dunders(self, device):
+        t = torch.randn((1,), device=device)
```
Comment (gchanan) [inline test/test_torch.py:5328]: how about testing inplace and, or, and shift operators.

**Review by gchanan — APPROVED** [review]

File: tools/autograd/templates/python_variable_methods.cpp (hunk 1 above)
Comment (mruberry) [reply tools/autograd/templates/python_variable_methods.cpp:857]: We don't. I made a note to follow-up.

File: test/test_torch.py (hunk 2 above)
Comment (mruberry) [reply test/test_torch.py:5328]: Good idea. I'll put those in a follow-up and like extend our tensor method tests to include these dunders, too.

**Pull request closed without merging**